src/System.Runtime.Numerics/src/System/Numerics/BigIntegerBuilder.cs

```diff
@@ -160,23 +160,11 @@ internal void GetIntegerParts(int signSrc, out int sign, out uint[] bits)
 
             int cuExtra = _rgu.Length - _iuLast - 1;
             Debug.Assert(cuExtra >= 0);
-            if (cuExtra <= 1)
+            if (cuExtra == 0)
             {
-                if (cuExtra == 0 || _rgu[_iuLast + 1] == 0)
```

Comment:
Why were you able to effectively eliminate half of this logic?  It was dead code?